src/main/java/sparksoniq/semantics/visitor/StaticContextVisitor.java

```diff
@@ -124,29 +91,24 @@ public StaticContext visitForClauseVar(ForClauseVar expression, StaticContext ar
                 expression.getMetadata()
             );
         }
-        // TODO visit at...
-        this.visit(expression.getExpression(), argument);
         return result;
     }
 
     @Override
     public StaticContext visitLetClauseVar(LetClauseVar expression, StaticContext argument) {
-        StaticContext result = visitFlowrVarDeclaration(expression, argument);
         this.visit(expression.getExpression(), argument);
```

Comment:
The resulting sc is not used? It is because this function has the side effect of storing the static context in the expression?

Reply:
clarified offline (visiting checks for variables)

Reply:
@ghislainfourny this pr needs approval as well